### ISSUE TITLE
[PATCH] perlfunc: another alternative for enabling the "state" feature

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#14372]
Migrated from [rt.perl.org#123505](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=123505) (status was 'resolved')

Searchable as RT123505$

**Comment by p5pRT** [comment]
### From @eserte

#### Created by @eserte



See attached patch\.

<details><summary>Perl Info</summary>

```

Flags:
    category=docs
    severity=wishlist
    Type=Patch
    PatchStatus=HasPatch

Site configuration information for perl 5.21.8:

Configured by eserte at Fri Dec 26 14:58:20 CET 2014.

Summary of my perl5 (revision 5 version 21 subversion 8) configuration:
  Commit id: 645419230fba91586f3efbb95fd62ad48b554375
  Platform:
    osname=freebsd, osvers=9.2-release, archname=amd64-freebsd
    uname='freebsd cvrsnica.herceg.de 9.2-release freebsd 9.2-release #0 r255898: thu sep 26 22:50:31 utc 2013 root@bake.isc.freebsd.org:usrobjusrsrcsysgeneric amd64 '
    config_args='-D useshrplib=true -Dprefix=/usr/perl5.21.7-93-g19bf100 -Dusemymalloc=n -D cc=ccache cc -D usedevel=define -Duse64bit -de -D hintfile=myfreebsd'
    hint=recommended, useposix=true, d_sigaction=define
    useithreads=undef, usemultiplicity=undef
    use64bitint=define, use64bitall=define, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='ccache cc', ccflags ='-DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -D_FORTIFY_SOURCE=2',
    optimize='-O',
    cppflags='-DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include'
    ccversion='', gccversion='4.2.1 20070831 patched [FreeBSD]', gccosandvers=''
    intsize=4, longsize=8, ptrsize=8, doublesize=8, byteorder=12345678, doublekind=3
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=16, longdblkind=3
    ivtype='long', ivsize=8, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='ccache cc', ldflags ='-Wl,-E  -fstack-protector -L/usr/local/lib'
    libpth=/usr/lib /usr/local/lib /usr/include/gcc/4.2 /usr/lib
    libs=-lgdbm -lm -lcrypt -lutil -lc -lpthread
    perllibs=-lm -lcrypt -lutil -lc -lpthread
    libc=, so=so, useshrplib=true, libperl=libperl.so
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='  -Wl,-R/usr/perl5.21.7-93-g19bf100/lib/5.21.8/amd64-freebsd/CORE'
    cccdlflags='-DPIC -fPIC', lddlflags='-shared  -L/usr/local/lib -fstack-protector'



@INC for perl 5.21.8:
    lib
    /usr/perl5.21.7-93-g19bf100/lib/site_perl/5.21.8/amd64-freebsd
    /usr/perl5.21.7-93-g19bf100/lib/site_perl/5.21.8
    /usr/perl5.21.7-93-g19bf100/lib/5.21.8/amd64-freebsd
    /usr/perl5.21.7-93-g19bf100/lib/5.21.8
    .


Environment for perl 5.21.8:
    HOME=/home/e/eserte
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH=/home/e/eserte/work2/perl
    LOGDIR (unset)
    PATH=/usr/local/bin:/usr/bin:/bin:/usr/local/sbin:/usr/sbin:/sbin:/home/e/eserte/bin/freebsd9.1:/home/e/eserte/bin/sh:/home/e/eserte/bin:/usr/games:/home/e/eserte/devel
    PERLDOC=-MPod::Perldoc::ToTextOverstrike
    PERL_BADLANG (unset)
    PERL_HTML_DISPLAY_CLASS=HTML::Display::Mozilla
    SHELL=/usr/local/bin/zsh

```
</details>

**Comment by p5pRT** [comment]
### From @eserte

<details><summary>0001-perlfunc-another-alternative-for-enabling-the-state-.patch</summary>

```diff
From 645419230fba91586f3efbb95fd62ad48b554375 Mon Sep 17 00:00:00 2001
From: Slaven Rezic <slaven@rezic.de>
Date: Fri, 26 Dec 2014 21:21:03 +0100
Subject: [PATCH] perlfunc: another alternative for enabling the "state"
 feature

---
 pod/perlfunc.pod | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/pod/perlfunc.pod b/pod/perlfunc.pod
index 387cccb..e7ccfa4 100644
--- a/pod/perlfunc.pod
+++ b/pod/perlfunc.pod
@@ -7701,7 +7701,8 @@ list context is currently not possible this would serve no purpose.
 
 C<state> variables are enabled only when the C<use feature "state"> pragma 
 is in effect, unless the keyword is written as C<CORE::state>.
-See also L<feature>.
+See also L<feature>. Alternately, include a C<use v5.10> or later to the
+current scope.
 
 =item study SCALAR
 X<study>
-- 
2.1.2


```
</details>

**Comment by p5pRT** [comment]
### From @cpansprout

On Fri Dec 26 12&#8203;:36&#8203;:08 2014\, slaven@&#8203;rezic\.de wrote&#8203;:
> This is a bug report for perl from slaven@&#8203;rezic\.de\,
> generated with the help of perlbug 1\.40 running under perl 5\.21\.8\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please describe your issue here\]
> 
> See attached patch\.

Thank you\.  Applied as 62fedd5149\.

\-\- 

Father Chrysostomos

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'open' to 'pending release'

**Comment by p5pRT** [comment]
### From @khwilliamson

Thanks for submitting this ticket

The issue should be resolved with the release today of Perl v5\.22\, available at http&#8203;://www\.perl\.org/get\.html
If you find that the problem persists\, feel free to reopen this ticket

\-\- 
Karl Williamson for the Perl 5 porters team

**Comment by p5pRT** [comment]
@khwilliamson - Status changed from 'pending release' to 'resolved'
